lib/routes/twreporter/fetch-article.ts

```diff
@@ -1,67 +1,99 @@
-    // 2. just a simple line starts with `文/ ...`
-    // For the first condition, we use a array
-    // to record the list of author(s), and use
-    // `；` to connect two lines.
-    const authors = [];
-
-    if (acquire('.loxoWO').text() === '') {
-        for (const item of acquire('.flciyI').get()) {
-            const $ = load(item);
-            const job = $('.hGsNtm').text();
-            // An article may have multiple authors
-            const name = [];
-            for (const item of $('.cidPTd > a').get()) {
-                const $ = load(item);
-                name.push($('.fJSaZP').text());
-            }
+    const banner_image = post.og_image.resized_targets.desktop.url;
+    const caption = post.leading_image_description;
+    const banner_description = post.og_image.description;
+    const og_description = post.og_description;
+    const banner = art(path.join(__dirname, 'templates/image.art'), { image: banner_image, description: banner_description, caption });
 
-            const author = job + '／' + name.join('，');
-            authors.push(author);
-        }
-    } else {
-        authors.push(acquire('.loxoWO').text());
-    }
-    const author = authors.join('；');
+    const text = post.content.api_data
+        .map((item) => {
+            const content = item.content;
+            const type = item.type;
+            let block = '';
+            if (content !== '' && type !== 'embeddedcode') {
+                switch (type) {
+                    case 'image':
+                    case 'slideshow':
+                        block = content.map((image) => art(path.join(__dirname, 'templates/image.art'), { image: image.desktop.url, description: image.description, caption: image.description })).join('<br>');
+
+                        break;
+
+                    case 'blockquote':
+                        block = `<blockquote>${content}</blockquote>`;
+
+                        break;
 
-    // contents = cover photo + italic intro + text
-    const contents = '<em>' + capture('.hxFBKc').html() + '</em><br>' + capture('.jONJYq').html();
+                    case 'header-one':
+                        block = `<h1>${content}</h1>`;
+
+                        break;
+
+                    case 'header-two':
+                        block = `<h2>${content}</h2>`;
+
+                        break;
+
+                    case 'infobox': {
+                        const box = content[0];
+                        block = `<h2>${box.title}</h2>${box.body}`;
+
+                        break;
+                    }
+                    case 'youtube': {
+                        const video = content[0].youtubeId;
+                        const id = video.split('?')[0];
+                        block = art(path.join(__dirname, 'templates/youtube.art'), { video: id });
+
+                        break;
+                    }
+                    default:
+                        block = `${content}<br>`;
+                }
```

Comment:
Please move the switch statement out of the map loop.

Reply:
Sorry, I'm a little confused, do you mean to use for loop instead? (Note that I have to use `switch(item.type)`.)

Reply:
> Please move the switch statement out of the map loop.

```diff
+ function switchFn() {}

const text = post.content.api_data
    .map(() => {
        ...
-        switch() {}
+        switchFn()
        ...
});